src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Debug/DebugProfileEnumValuesGenerator.cs

```diff
@@ -0,0 +1,81 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Build.Framework.XamlTypes;
+using Microsoft.VisualStudio.Threading;
+using Microsoft.VisualStudio.ProjectSystem.Properties;    
+
+namespace Microsoft.VisualStudio.ProjectSystem.Debug
+{
+
+    /// <summary>
+    /// Provides the IEnumValue's for the ActiveDebugProfile property. This is what is used to drive
+    /// the debug target selection dropdown.
+    /// </summary>
+
+    internal class DebugProfileEnumValuesGenerator : IDynamicEnumValuesGenerator
+    {
+        /// <summary>
+        /// The listed values.
+        /// </summary>
+        private AsyncLazy<ICollection<IEnumValue>> listedValues;
+
+        /// <summary>
+        /// Create a new instance of the class.
+        /// </summary>
+        internal DebugProfileEnumValuesGenerator(ILaunchSettingsProvider profileProvider)
+        {
+            this.listedValues = new AsyncLazy<ICollection<IEnumValue>>(delegate 
+            {
+                var curSnapshot = profileProvider.CurrentSnapshot;
+                if (curSnapshot != null )
+                {
+                    return Task.FromResult(GetEnumeratorEnumValues(curSnapshot));
+                }
+                ICollection<IEnumValue> emptyCollection = new List<IEnumValue>();
+                return Task.FromResult(emptyCollection);
+            });
+        }
+
+        /// <summary>
+        /// See <see cref="IDynamicEnumValuesGenerator"/>
+        /// </summary>
+        public async Task<ICollection<IEnumValue>> GetListedValuesAsync()
+        {
+            return await listedValues.GetValueAsync().ConfigureAwait(true);
+        }
+
+        /// <summary>
+        /// See <see cref="IDynamicEnumValuesGenerator"/>
+        /// </summary>
+        public bool AllowCustomValues
+        {
+            get { return false; }
+        }
+
+        /// <summary>
+        /// See <see cref="IDynamicEnumValuesGenerator"/>
+        /// </summary>
+        public async Task<IEnumValue> TryCreateEnumValueAsync(string userSuppliedValue)
+        {
+            return (await listedValues.GetValueAsync().ConfigureAwait(true))
+            .FirstOrDefault(v => LaunchProfile.IsSameProfileName(v.Name, userSuppliedValue));
+        }
+
+        internal static ICollection<IEnumValue> GetEnumeratorEnumValues(ILaunchSettings profiles)
```

Comment:
I am going to leave it. The IDynamicEnumValuesGenerator has methods which expect ICollection.